pints/_mcmc/_adaptive_covariance.py

```diff
@@ -58,21 +65,6 @@ def ask(self):
         if not self._running:
             self._initialise()
 
-        # Propose new point
```

Comment:
So... the base class has an ask(), and the implementing classes have to override it, before calling the base class ask() , doing their own stuff and actually returning.

That's OK. But wondering if it would be tidier to have the base class call a 2nd method, e.g. _ask() or _propose() that the base class defines but sticks a `raise NotImplementedError` in. That way we have made it more explicit that this is something implementing classes need to do?

Not sure! Thoughts @martinjrobins @ben18785 ?

Reply:
I like doing it the way we currently are because it avoids second methods and also, to me at least, this seems like a good way to use the inheritance of base classes. @martinjrobins what do you think?

Reply:
I'm happy with what @ben18785 has done, its a fairly traditional way of using inheritance.